cmd/p2p/p2p.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"fmt"
+	"github.com/spacemeshos/go-spacemesh/api"
+	cmdp "github.com/spacemeshos/go-spacemesh/cmd"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/metrics"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+// Cmd is the p2p cmd
+var Cmd = &cobra.Command{
+	Use:   "p2p",
+	Short: "start p2p",
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("Starting p2p")
+		p2pApp := NewP2PApp()
+		defer p2pApp.Cleanup()
+
+		p2pApp.Initialize(cmd)
+		p2pApp.Start(cmd, args)
+	},
+}
+
+func init() {
+	cmdp.AddCommands(Cmd)
+}
+
+type P2PApp struct {
```

Comment:
Why not remove the BaseApp? Then the Hare can be composed of P2PApp. Same goes for the SpacemeshApp.

Reply:
If we'll do that it will be a problem compiling the p2p app as a standalone

Reply:
Because it has to have a `main` to do so. and if we'll do it we won't be able to use it embedded in other structs.